src/pages/Auth/Login.jsx

```diff
@@ -1,59 +1,60 @@
 import React from "react";
-import { Link } from "react-router-dom";
+import { Link, useLocation, useNavigate } from "react-router-dom";
 import styles from "./Auth.module.css";
 import { useAuthForm, useDocumentTitle, useToggle } from "../../hooks";
 import { TextInput } from "../../components";
-import { login, axiosRequest } from "../../services";
+import { login } from "../../services";
 import { ImSpinner9 } from "react-icons/im";
 import { useAuth } from "../../contexts/providers/AuthProvider";
 import { EMAIL_REGEX } from "../../utils";
+import { toast } from "react-toastify";
 
 function Login() {
   const { authFormState: formState, authFormDispatch: formDispatch } =
     useAuthForm();
 
   const { authDispatch } = useAuth();
 
+  const location = useLocation();
+  const navigate = useNavigate();
+
   const { toggle: loginToggle, setToggle: setLoginToggle } = useToggle(false);
   const { toggle: guestLoginToggle, setToggle: setGuestLoginToggle } =
     useToggle(false);
 
   const { email, password } = formState;
 
-  const loginHandler = (e) => {
+  const loginHandler = async (e) => {
     e.preventDefault();
 
     if (password.length === 0) {
-      alert("Fields can't be empty"); //Need to change to custom alert component
+      toast.info("Fields can't be empty"); //Need to change to custom alert component
       return;
     }
 
     if (!EMAIL_REGEX.test(email)) {
-      alert("Invalid email address");
+      toast.info("Invalid email address");
       return;
     }
 
-    login(
-      {
-        email,
-        password,
-      },
-      setLoginToggle,
-      authDispatch
-    );
+    await login({ email, password }, setLoginToggle, authDispatch);
+    const pathName = location?.state?.from?.pathname;
+    pathName ? navigate(pathName) : navigate("/");
```

Comment:
just a small suggestion:
navigate(pathName ? pathName : "/"), this would have been easier to read.